### PR TITLE
Remove some compiler warnings.

### PR DESCRIPTION
**Pull request by sergeisakov**

### CODE REVIEW

File: lib/simulator_avx.h [hunk 1]
```diff
@@ -153,6 +153,7 @@ class SimulatorAVX final {
       break;
     default:
       // Cannot reach here.
+      ml = _mm256_set_epi32(0, 0, 0, 0, 0, 0, 0, 0);
```
Comment (95-martin-orion) [inline lib/simulator_avx.h:156]: It's probably preferable to raise an error here, if we have a way to do so.

(Similarly for other "unreachable" sections in this file)

File: lib/statespace.h [hunk 2]
```diff
@@ -48,8 +48,11 @@ class StateSpace {
       return State((fp_type*) _aligned_alloc(vector_size, 64), &_aligned_free);
     #else
       void* p = nullptr;
-      posix_memalign(&p, 64, vector_size);
-      return State((fp_type*) p, &free);
+      if (posix_memalign(&p, 64, vector_size) == 0) {
+        return State((fp_type*) p, &free);
+      } else {
+        return State(nullptr, &free);
```
Comment (95-martin-orion) [inline lib/statespace.h:54]: Should this be an error, or is it a valid State object?

File: lib/simulator_avx.h (hunk 1 above)
Comment (sergeisakov) [reply lib/simulator_avx.h:156]: How do you propose to raise an error here? These sections are really unreachable if those functions are used correctly. The correct usage is described in function comments.

File: lib/statespace.h (hunk 2 above)
Comment (sergeisakov) [reply lib/statespace.h:54]: No, it is an invalid State object. The validity can be tested by the `IsNull()` function as it is done in `run_qsim.h` and `hybrid.h`.

File: lib/simulator_avx.h (hunk 1 above)
Comment (95-martin-orion) [reply lib/simulator_avx.h:156]: Sorry, I was in a Python state of mind. This should be fine as-is.

File: lib/statespace.h (hunk 2 above)
Comment (95-martin-orion) [reply lib/statespace.h:54]: Okay. As long as it is caught somewhere, this is fine.

**Pull request merged**